spec/paperclip/attachment_spec.rb

```diff
@@ -1477,6 +1478,53 @@ def call(filename)
     end
   end
 
+  context "an attachment with process_in_background set" do
+    before do
+      ActiveJob::Base.queue_adapter = :test
+      ActiveJob::Base.logger = nil
+
+      rebuild_model styles: { thumb: "100x100" },
+                    only_process: [:none],
+                    process_in_background: [:thumb]
+      @file = File.new(fixture_file("5k.png"), "rb")
+      GlobalID.app = "paperclip-test"
+      Dummy.send(:include, GlobalID::Identification)
+      @dummy = Dummy.new
+    end
+
+    after do
+      ActiveJob::Base.queue_adapter.enqueued_jobs.clear
+    end
+
+    it "enqueues job to process given styles in background" do
+      @dummy.avatar = @file
+      @dummy.save!
+      jobs = ActiveJob::Base.queue_adapter.enqueued_jobs
+
+      assert_equal 1, jobs.count
+
+      job = jobs[0]
+      assert_equal Paperclip::ProcessJob, job[:job]
+      assert_equal @dummy, GlobalID::Locator.locate(job[:args][0].values[0])
+      assert_equal "avatar", job[:args][1]
+      assert_equal "default", job[:queue]
+    end
+
+    it "does not process given styles in real time" do
+      attachment = @dummy.avatar
+      attachment.expects(:post_process).with(:none)
```

Comment:
I'm little confused when to use rspec and when minitest - which one is preferred?